File: programs/ifactor_1000000000000037/Erlang-trial-odd_BEAM/ifactor.erl
% Copyright (C) 2017, Florent Gallaire <fgallaire@gmail.com>
% MIT license

-module(ifactor).
-export([main/1]).

ifactor(N) when is_integer(N), N > 1 -> ifactor(N, N div 2, N rem 2, 2, []).
ifactor(N, Dk, P) when N > 1 -> ifactor(N, N div Dk, N rem Dk, Dk, P);
ifactor(_N, _Dk, P) -> lists:reverse(P).
ifactor(_N, Q, 0, Dk, P) -> ifactor(Q, Dk, [Dk|P]);
ifactor(N, Q, _R, 2, P) when Q > 2 -> ifactor(N, 3, P);
ifactor(N, Q, _R, Dk, P) when Q > Dk -> ifactor(N, Dk + 2, P);
ifactor(N, _Q, _R, _Dk, P) -> lists:reverse([N|P]).

main([String]) ->
    N = list_to_integer(String),
    P = ifactor(N),
    io:format("~w: ~s\n", [N, string:join([integer_to_list(I) || I <- P], " ")]),
    erlang:halt(0).

File: programs/ifactor_1000000000000037/Python3-trial-odd_CPython/ifactor.py
# Copyright (C) 2017, Florent Gallaire <fgallaire@gmail.com>
# MIT license

from sys import argv

N = argv[1]
n = int(N)
d = iter(range(3, n, 2))
dk = 2
p = []

while n > 1:
    q, r = divmod(n, dk)
    if r == 0:
        p.append(dk)
        n = q
    elif q > dk:
        dk = next(d)
    else:
        p.append(n)
        break

print(N + ': ' + ' '.join(str(i) for i in p))
